state/mongodb/mongodb.go

```diff
@@ -179,6 +179,8 @@ func (m *MongoDB) setInternal(ctx context.Context, req *state.SetRequest) error
 	filter := bson.M{id: req.Key}
 	if req.ETag != nil {
 		filter[etag] = *req.ETag
+	} else if req.Options.Concurrency == state.FirstWrite {
+		filter[etag] = uuid.NewString()
```

Comment:
Was this tested? Why matching eTag on a random UUID fixes the firstWrite wins?

Reply:
Tested, if set etag when firstWrite, then the second write would raise a error.